add creator functions (create/extend) to ViewBridge.js

chee opened the pull request:
these are a pair of little friendly functions to make making ViewBridges arguably a little nicer to create.

a ViewBridge creator looks like this:

``` javascript
rhubarb.vb.create('WowMyViewBridge', function(parent) {
  const Wow = this // reference to static class
  this.staticMethod = function(hamster, dog) {
    // me oh my my my
  }

  return {
    instanceProperty: true,
    instanceMethod() {
      // what what
    },
    get instanceProperty() { /* do something */ },
    set instanceProperty(val) { /* wow another thing */ },
    handleInput() { /* magic lights */ },
    onReady() {
      this.certainElement = this.viewNode.getElementsByClassName('js-element')[0]
      this.certainElement.addEventListener('click', event => event.target.background = '#ff2a50')
      this.viewNode.addEventListener('input', this.handleInput)
    }
  }
})
```

chee committed:
add creator functions (create/extend) to ViewBridge.js

acuthbert commented:
Sweet! I haven't tried it yet, one comment, how would you feel if we added an alias on window for vb to View bridge so you could just do

``` js
rhubarb.viewBridgeClasses.WowMyViewBridge = vb.create(function() {

})
```

Taking it even further, what if the registration of the view bridge became easier too:

``` js
vb.create('WowMyViewBridge',function() {

})
```

This way there's no long 'namespace' objects to refer to at all. Does that work?

chee committed:
take a name argument in ViewBridge.create

chee committed:
pass parent prototype in to view bridge creator

chee committed:
add viewbridge shorthand

chee commented:
@acuthbert cool, yes. i've updated the description to show what it looks like when it works like that.

i've also added `parent.prototype` as the first argument to the ViewBridge creator, so you can do:

``` javascript
onReady: function() {
  parent.onReady.call(this);
  /* do something fancy */
}
```

that could still be made nicer by perhaps binding all the functions on the parent to the child during the extend, but i'm not sure if that's always the desired behaviour
